doc/stages/filters.hag.rst

```diff
@@ -122,23 +127,44 @@ or :ref:`filters.smrf` to label ground returns, as shown
           "type":"filters.smrf"
       },
       {
-          "type":"filters.hag"
+          "type":"filters.hag",
+          "count":4
       },
       {
           "type":"filters.ferry",
-          "dimensions":"HeightAboveGround=Z"
+          "dimensions":"HeightAboveGround=>Z"
       },
       "autzen-height-as-Z-smrf.laz"
   ]
 
-which is once again equivalent to the command
+which is equivalent to the command
 
 ::
 
     $ pdal translate autzen.laz autzen-height-as-Z-smrf.bpf smrf hag ferry \
-        --filters.ferry.dimensions="HeightAboveGround=Z"
+        --filters.ferry.dimensions="HeightAboveGround=Z" \
+        --filters.hag.count=4
 
 Options
 -------------------------------------------------------------------------------
 
-None
+_`count`
+    The number of ground neighbors to consider when determining the height
+    above ground for a non-ground point.  Must be at least 3 when using
+    the `delaunay`_ option. [Default: 1]
+
+_`delaunay`
+    If true, create a delaunay triangulation of neighboring ground points
+    to aid in interpolation of a ground height to use as a difference to
+    a point's non-ground height. [Default: false]
+
+max_distance
+    Eliminate non-ground points from consideration when their distance
+    from a ground point exceeds this amount.  Not used with the `delaunay`_
```

Comment:
I'm not sure this is actually what we are doing, whether or not it's what we intend to do. The way I'm reading the code, the loop over the ground neighbors breaks whenever the threshold is exceeded, but any neighbors encountered up until that point still impact the result.

Reply:
But the neighbors come in order.  Closest first...

Reply:
Right, but if you look at that section of code, as I'm reading it, the first neighbors may very well be under the max distance, and will begin to update `z_accumulator` and `weights`. When a neighbor exceeds the distance, it bails and does the final update to `z1` based on the values accumulated up to that point. Maybe the wording is just backwards and should be "Eliminate ground points from consideration when their distance from a non-ground point exceeds..."? The way it is written now could lead one to believe that non-ground points will not have HAG computed if the ground neighbors are too far away (whether that's the nearest ground point, or just any of the neighbors is unclear).

Reply:
Yes, backwards.  Do you have better wording?  I hate to get too verbose.

Reply:
Perhaps something like

> Use default Z value whenever any neighbors in the ground set exceeds the maximum distance to the non-ground point.

Reply:
Based off our other conversation on the topic

> Use only ground points within `max_distance` of non-ground point when performing neighbor interpolation.